lib/api/2.0/nodes.js

```diff
@@ -78,9 +77,10 @@ var nodesGetWorkflowById = controller(function(req) {
         newQuery = req.query;
     }
 
-    return workflowApiService.getWorkflowsByNodeId(req.swagger.params.identifier.value,
-                                                   newQuery);
-
+    return waterline.nodes.needByIdentifier(req.swagger.params.identifier.value)
```

Comment:
@lacarb It looks like this query is unnecessary since `nodes.getNodeWorkflowById` already calls `waterline.nodes.needByIdentifer` here: https://github.com/RackHD/on-http/blob/master/lib/services/nodes-api-service.js#L544